bindings/java/src/org/sleuthkit/datamodel/CommunicationsManager.java

```diff
@@ -1409,16 +1379,102 @@ private String normalizePhoneNum(String phoneNum) {
 	 * Utility method to convert a list to an CSV string
 	 */
 	private static <T> String buildCSVString(Collection<T> values) {
-		if (values==null || values.isEmpty()) return "";
+		if (values==null || values.isEmpty()) 
+			return "";
+		
 		StringBuilder result = new StringBuilder();
 		for (T val : values) {
 			result.append(val);
 			result.append(",");
 		}
+		
 		return result.substring(0, result.length() - 1);
 	}
 	
-	/*
+	/**
+	 * Builds the SQL for the given CommunicationsFilter.
+	 * 
+	 * Gets the SQL for each subfilter and combines using AND 
+	 */
+	private String getCommunicationsFilterSQL(CommunicationsFilter commFilter) {
+		if ( null == commFilter || commFilter.getAndFilters().isEmpty() )
+			return "";
+		
+		StringBuilder sqlSB = new StringBuilder();
+		boolean first = true;
+		for (SubFilter subFilter: commFilter.getAndFilters()) {
+			if (first) {
+				first = false;
+			}
+			else {
+				sqlSB.append(" AND ");
+			}
+			sqlSB.append("( ");
+			sqlSB.append(getSubFilterSQL(subFilter));
+			sqlSB.append(" )");
+		}	
+		
+		return "( " + sqlSB.toString() + " )";
+	}
+	
+	/**
+	 * Returns the SQL for a subfilter.
+	 */
+	private String getSubFilterSQL(SubFilter subFilter) {
+		String result = "";
+		
+        if (subFilter == null) {
+            return "";
+        } else if (subFilter instanceof DeviceFilter) {
+            result = getSQL((DeviceFilter) subFilter);
+        } else if (subFilter instanceof AccountTypeFilter) {
```

Comment:
Given that it looks like the getSQL methods can be static, is it possible to add a getSQL method to the SubFilter class and thereby eliminate the need for casting here?